### PR TITLE
Add opencv detection api call on rtmp publish 

### PR DESCRIPTION
**Pull request by melvinheks**

### CODE REVIEW

**Comment by melvinheks** [comment]
#100 pr heh

File: src/web/server/app/tests/controllers/auth.test.ts [hunk 1]
```diff
@@ -620,8 +628,46 @@ describe('auth', () => {
...
+    });
+    it('should return 500 on a unsuccessful opencv detection call', async () => {
+      mockedAxios.get.mockImplementationOnce(() =>
+        Promise.resolve({ status: 200 })
+      );
+      mockedAxios.get.mockImplementationOnce(() =>
+        Promise.resolve({ status: 400 })
+      );
+      const mockResults = {
+        rows: [{ session: JSON.stringify({ userId: 'wrry' }) }],
```
Comment (yliu0715) [inline src/web/server/app/tests/controllers/auth.test.ts:644]: It's wrrryaaaah

File: src/web/server/app/controllers/auth.ts [hunk 2]
```diff
@@ -291,7 +294,21 @@ const rtmpAuthPublish = async (req: Request, res: Response, start: boolean) => {
         );
         const saverResponse = await axios.get(saverUrl);
         if (saverResponse.status === 200) {
-          res.status(200).send('OK');
+          if (start) {
```
Comment (yliu0715) [inline src/web/server/app/controllers/auth.ts:297]: What's `start` ?

File: src/web/server/app/controllers/auth.ts [hunk 3]
```diff
@@ -291,7 +294,21 @@ const rtmpAuthPublish = async (req: Request, res: Response, start: boolean) => {
...
+                `?camera_id=${req.body.cameraId}&stream_id=${req.body.name}`
+            );
+            const cvResponse = await axios.get(cvUrl);
+            if (cvResponse.status === 200) {
+              res.status(200).send('OK');
+            } else {
+              res.status(500).send('Server error');
+            }
+          } else {
+            res.status(200).send('OK');
```
Comment (yliu0715) [inline src/web/server/app/controllers/auth.ts:310]: How come you send a 200 here and also send 200 in the if (start) statement?

File: src/web/server/app/tests/controllers/auth.test.ts (hunk 1 above)
Comment (melvinheks) [reply src/web/server/app/tests/controllers/auth.test.ts:644]: yes

File: src/web/server/app/controllers/auth.ts (hunk 2 above)
Comment (melvinheks) [reply src/web/server/app/controllers/auth.ts:297]: start indicates whether the rtmp stream has started or ended

Comment (yliu0715) [reply src/web/server/app/controllers/auth.ts:297]: yes

File: src/web/server/app/controllers/auth.ts (hunk 3 above)
Comment (yliu0715) [reply src/web/server/app/controllers/auth.ts:310]: ok

**Pull request merged**